qiskit/pulse/schedule.py

```diff
@@ -1025,12 +1147,51 @@ def instructions(self) -> Tuple[Tuple[int, Instruction]]:
     @property
     def blocks(self) -> Tuple["BlockComponent", ...]:
         """Get the time-ordered instructions from self."""
-        return tuple(self._blocks)
+        return tuple(self._blocks.nodes())
+
+    def _collect_parameters(self, scope: str, add_scope: bool = False) -> Set[Parameter]:
```

Comment:
I wonder if `scope` should be `scope: Tuple[str, ...]` instead to avoid reserving `.` as a special character.